digest/cache.go

```diff
@@ -61,13 +61,13 @@ func (c *Cache) Get(ctx context.Context, file paranoid.File) (ForUpload, error)
 		return result, nil
 	case cache.NotFound, cache.DoNotPromote:
 	default:
-		return ForUpload{}, getErr
+		return forUpload(), getErr
```

Comment:
Is there any reason we can't return nil on error? ForUpload is an interface now. Unless we still need it to be created if we get an error.